name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DABPDFExtractor.java

```diff
@@ -1221,6 +1253,138 @@ private void addAccountStatementTransaction()
+                            accountTransaction.setType(AccountTransaction.Type.FEES);
+                            return accountTransaction;
+                        })
+
+                        .section("amount", "date", "note") //
+                        .documentContext("currency", "year") //
+                        .match("^([\\d]{2}\\.[\\d]{2}\\.) (?<date>[\\d]{2}\\.[\\d]{2}\\.) (Verwalterpreis) (?<amount>[\\.,\\d]+)\\-$") //
+                        .match("^(?<note>.*) End to End\\-") //
+                        .assign((t, v) -> {
+                            t.setDateTime(asDate(v.get("date") + v.get("year")));
+                            t.setCurrencyCode(v.get("currency"));
+                            t.setAmount(asAmount(v.get("amount")));
+                            t.setNote(v.get("note"));
+                        })
+
+                        .wrap(TransactionItem::new));
+
+        // @formatter:off
+        // 07.11. 11.11. Wertpapierverkauf 0,49+
+        // VKF    A2061191320241107   IE00B52MJY50*         0,003
+        //
+        // 27.09. 01.10. Wertpapierkauf 6,26-
+        // KAUF   A6239874820240927   IE00BL25JM42*         0,150
+        // @formatter:on
+        Block BuySellBlock = new Block(
+                        "^([\\d]{2}\\.[\\d]{2}\\.) ([\\d]{2}\\.[\\d]{2}\\.) (Wertpapierverkauf|Wertpapierkauf) ([\\.,\\d]+)[\\+\\-]$");
+        type.addBlock(BuySellBlock);
+        BuySellBlock.set(new Transaction<BuySellEntry>()
+
+                        .subject(() -> {
+                            BuySellEntry portfolioTransaction = new BuySellEntry();
+                            portfolioTransaction.setType(PortfolioTransaction.Type.BUY);
+                            return portfolioTransaction;
+                        })
+
+                        .section("amount", "date", "isin", "shares", "type") //
+                        .documentContext("currency", "year") //
+                        .match("^([\\d]{2}\\.[\\d]{2}\\.) (?<date>[\\d]{2}\\.[\\d]{2}\\.) (?<type>Wertpapierverkauf|Wertpapierkauf) (?<amount>[\\.,\\d]+)[\\+\\-]$") //
+                        .match("^.* (?<isin>[A-Z]{2}[A-Z0-9]{9}[0-9])\\* +(?<shares>[\\.,\\d]+)$") //
+                        .assign((t, v) -> {
+                            if ("Wertpapierverkauf".equals(v.get("type")))
+                                t.setType(PortfolioTransaction.Type.SELL);
+                            t.setSecurity(getOrCreateSecurity(v));
+                            t.setDate(asDate(v.get("date") + v.get("year")));
+                            t.setCurrencyCode(v.get("currency"));
+                            t.setShares(asShares(v.get("shares")));
+                            t.setAmount(asAmount(v.get("amount")));
+                            v.getTransactionContext().put(FAILURE,
+                                            Messages.MsgErrorTransactionAlternativeDocumentRequired);
+                        })
```

Comment:
As you know :wink:, we do not import securities bookings via account statements. In your example, you have already shown via Kauf14.txt that there are separate statements.

Reply:
Somehow they generate account statements with buys only. To have a better error message I tried it this way. Tagging it as non importance.